app/models/manageiq/providers/hawkular/inventory/parser/availability_updates.rb

```diff
@@ -3,22 +3,34 @@ class Hawkular::Inventory::Parser::AvailabilityUpdates < ManagerRefresh::Invento
     def parse
       fetch_server_availabilities
       fetch_deployment_availabilities
+      fetch_domain_availabilities
     end
 
     private
 
+    def find_updated_resource(resource_name)
+      collector.send("#{resource_name}_updates").each do |item|
+        resource = persister.send("middleware_#{resource_name.to_s.pluralize}").find_or_build(item.manager_ref[:ems_ref])
+        yield(resource, item)
+      end
+    end
+
     def fetch_server_availabilities
-      collector.server_updates.each do |item|
-        server = persister.middleware_servers.find_or_build(item.manager_ref[:ems_ref])
+      find_updated_resource(:server) do |server, item|
         server.properties = item.options
       end
     end
 
     def fetch_deployment_availabilities
-      collector.deployment_updates.each do |item|
-        deployment = persister.middleware_deployments.find_or_build(item.manager_ref[:ems_ref])
+      find_updated_resource(:deployment) do |deployment, item|
         deployment.status = item.options[:status]
       end
     end
+
+    def fetch_domain_availabilities
+      find_updated_resource(:domain) do |domain, item|
```

Comment:
Is this being persisted somewhere? Seems like it is only updating the object but not persisting it.

Reply:
In this part we are not persisting, just parsing data. There is a persister which is in charge of it.